src/cargo/util/lints.rs

```diff
@@ -0,0 +1,226 @@
+use crate::core::FeatureValue::Dep;
+use crate::core::{Edition, FeatureValue, Package};
+use crate::util::interning::InternedString;
+use crate::{CargoResult, GlobalContext};
+use annotate_snippets::{Level, Renderer, Snippet};
+use cargo_util_schemas::manifest::{TomlLintLevel, TomlToolLints};
+use pathdiff::diff_paths;
+use std::collections::HashSet;
+use std::ops::Range;
+use std::path::Path;
+use toml_edit::ImDocument;
+
+fn get_span(document: &ImDocument<String>, path: &[&str], get_value: bool) -> Option<Range<usize>> {
```

Comment:
Forgot to review this part during last round...

Not sure if this kind of access still have the correct span, but at least it passes all new tests :)

```rust
fn get_span(
    document: &ImDocument<String>,
    paths: &[&str],
    get_value: bool,
) -> Option<Range<usize>> {
    let (last, paths) = paths.split_last()?;
    let mut container = document.as_item();
    for path in paths {
        container = container.get(path)?;
    }
    if get_value {
        container.get(last)?.span()
    } else {
        container.as_table()?.key(last)?.span()
    }
}
```

Reply:
Sadly, I don't think this works for `Value::Array` as `item.get()` only works on [`Item::Table` and `Value::InlineTable`](https://github.com/toml-rs/toml/blob/bb92dc30c2a3786d583b6d4a27c345b31c232892/crates/toml_edit/src/index.rs#L36-L51) (I think)